[No Merge] Sandbox to test the migration from Blockstore to LC

kdmccormick opened the pull request:
### PR sandbox

**Settings**
```yaml
PLUGINS:
    - mfe
    - grove
    - s3
    - library-authoring-mfe
```
**Tutor requirements**
```
git+https://github.com/overhangio/tutor.git@nightly
git+https://github.com/overhangio/tutor-mfe.git@nightly
git+https://gitlab.com/opencraft/dev/tutor-contrib-grove.git@main
git+https://github.com/hastexo/tutor-contrib-s3.git@main
git+https://github.com/openedx/openedx-tutor-plugins@main#subdirectory=plugins/tutor-contrib-library-authoring-mfe
```
______________________

This PR started out as a no-op so that a master-equivalent sandbox was built. We then push all the changes from https://github.com/openedx/edx-platform/pull/34066 in order to confirm that the migration runs OK.

ormsbee committed:
feat: convert v2 content libraries to use Learning Core models

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at https://pr-33936-7daceb.sandboxes.opencraft.hosting
Sandbox Studio is available at https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5788144660-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5788144660-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5788144660-tutor-requirements.txt

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at https://pr-33936-7daceb.sandboxes.opencraft.hosting
Sandbox Studio is available at https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5803879351-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5803879351-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5803879351-tutor-requirements.txt

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at https://pr-33936-7daceb.sandboxes.opencraft.hosting
Sandbox Studio is available at https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5804250619-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5804250619-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5804250619-tutor-requirements.txt

open-craft-grove commented:
Sandbox deployment failed.

Check failure logs here https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5878648060.log

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5878648060-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5878648060-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5878648060-tutor-requirements.txt

Please check the settings and requirements and retry deployment by updating the pull request description or pushing a new commit

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at https://pr-33936-7daceb.sandboxes.opencraft.hosting
Sandbox Studio is available at https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5878648053-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5878648053-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5878648053-tutor-requirements.txt

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at https://pr-33936-7daceb.sandboxes.opencraft.hosting
Sandbox Studio is available at https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5879753516-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5879753516-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-5879753516-tutor-requirements.txt

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at https://pr-33936-7daceb.sandboxes.opencraft.hosting
Sandbox Studio is available at https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6085467787-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6085467787-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6085467787-tutor-requirements.txt

ormsbee committed:
refactor: no need to spell out the app config when there's only one

ormsbee committed:
chore: remove comment about libraries being blockstore-based

ormsbee committed:
test: fix model tests for content_libraries

ormsbee committed:
test: remove unnecessary mixin layering of test; remove parent/child tests

(we don't support that)

ormsbee committed:
fix: fix the filtering test case (requires change from openedx-learning)

ormsbee committed:
test: remove links-related tests (we no longer have them)

ormsbee committed:
test: remove tests regarding library-type conversion (we're no longer supporting that).

ormsbee committed:
test: more text patching

ormsbee committed:
fix: fix default xblock title behavior and adjust more tests

ormsbee committed:
fix: deleted blocks should 404 instead of 500

ormsbee committed:
feat!: remove blockstore lib import to learning core

There are no live v2 blockstore libraries that we're planning to
migrate.

ormsbee committed:
refactor: adjust openedx-learning usage to match 0.6.0 changes

ormsbee committed:
test: fix content_tagging tests to remove blockstore collection use

open-craft-grove commented:
Sandbox deployment failed.

Check failure logs here https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6139973838.log

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6139973838-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6139973838-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6139973838-tutor-requirements.txt

Please check the settings and requirements and retry deployment by updating the pull request description or pushing a new commit

ormsbee committed:
test: remove test for v2 lib child block functionality (now unsupported)

ormsbee committed:
test: remove blockstore usage from clipboard paste test

open-craft-grove commented:
Sandbox deployment failed.

Check failure logs here https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6143461438.log

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6143461438-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6143461438-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6143461438-tutor-requirements.txt

Please check the settings and requirements and retry deployment by updating the pull request description or pushing a new commit

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at https://pr-33936-7daceb.sandboxes.opencraft.hosting
Sandbox Studio is available at https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6143486262-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6143486262-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6143486262-tutor-requirements.txt

ormsbee committed:
fix: remove atomic decorator from views

ormsbee committed:
refactor: shift library search logic from view to api

ormsbee committed:
fix: force content libraries views to be nonatomic

ormsbee committed:
refactor: remove unnecessary imports

ormsbee committed:
fix: have field data properly track changed fields (and simplify the runtime/fielddata)

ormsbee committed:
fix: remove error log that I was only using for debugging purposes

ormsbee committed:
refactor: cleanups, comments

ormsbee committed:
refactor: more comment tweaks

ormsbee committed:
refactor: more cleanup of config and removing debug comments

ormsbee committed:
refactor: more cleanup

ormsbee committed:
refactor: remove unused imports

ormsbee committed:
refactor: update comments

ormsbee committed:
refactor: remove unnecessary atomic block

ormsbee committed:
refactor: add comments to api calls, remove unused function

ormsbee committed:
refactor: changes in response to Kyle's review

ormsbee committed:
refactor: consolidate get_component_from_usage_key implementations

ormsbee committed:
refactor: comment

ormsbee committed:
refactor: remove get_block_display_name

It was previously being called with XBlocks or usage keys, at which
point, it would do a lookup of the display name from the Component. But
now it's only being sent XBlocks, and it only grabs their display_name
when that's the case, so there's no point in keeping it around.

ormsbee committed:
refactor: remove unnecessary config value for OPENEDX_LEARNING

ormsbee committed:
refactor: remove ENABLE_LIBRARY_AUTHORING_MICROFRONTEND from devstack settings

(you have to add it to private.py)

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at https://pr-33936-7daceb.sandboxes.opencraft.hosting
Sandbox Studio is available at https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting

Instance Tutor config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6173869287-tutor-config.yml
Instance Grove config : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6173869287-grove-config.yml
Instance Python requirements : https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6173869287-tutor-requirements.txt

ormsbee committed:
fix: make schema migration more backwards-compatible

ormsbee committed:
fix: add LC models to keyword overrides (I'm checking now if this is okay)

ormsbee committed:
fix: address linter errors

ormsbee committed:
fix: more linter fixes

ormsbee committed:
fix: more linter fixes

ormsbee committed:
fix: all the linter things

ormsbee committed:
fix: spacing

ormsbee committed:
fix: more linter fixes for indentation/spacing

ormsbee committed:
fix: all the sytax linter stuff, I think...

ormsbee committed:
fix: remove entries from db_keyword_overrides (no longer necessary)

ormsbee committed:
chore: mypy thinks it's a BaseManager and not a ContentLibraryManager

ormsbee committed:
fix: skip the libraries test cases under LMS (signals won't get sent)

ormsbee committed:
refactor: re-order skip and ddt decorators in case that makes any difference

kdmccormick committed:
refactor: address some minor review comments

* specify 'draft' in docstring for get_library_block
* make create_component_for_block private
* delete leftover get_learning_context_impl comment

kdmccormick committed:
refactor: get_library_block_olx -> get_block_draft_olx

kdmccormick committed:
test: use OpenEdxEventsTestMixin to fix content_libraries event testing

I'm not going to pretend I entirely understand this, but I'd like to get
tests passing first, and then worry about the details.

kdmccormick committed:
fix: raise a ValueError if block is set to OLX with the wrong XML tag

kdmccormick committed:
fix: wrap _create_component_for_block in a transaction

kdmccormick committed:
test: use non-ascii characters in library org, slug, and block id

kdmccormick committed:
refactor: don't import MediaType from models

kdmccormick committed:
fix: fall back to type-defined block name if display_name missing

kdmccormick committed:
fix: typo in get_block_display_name

kdmccormick committed:
test: fix non-ascii content_libraries tests

kdmccormick committed:
refactor: component_already_exists -> _component_exists

kdmccormick committed:
docs: add TODO comments - draft vs published, and deserialization

kdmccormick committed:
Merge remote-tracking branch 'upstream/master' into learning-core-libs-readwrite-runtime

kdmccormick committed:
feat: allow non-ascii block ids

kdmccormick committed:
feat!: rm get_library_blocks

kdmccormick committed:
Merge remote-tracking branch 'upstream/master' into learning-core-libs-readwrite-runtime

kdmccormick committed:
fix: typo -- left something commented out while testing

kdmccormick committed:
fix: `only` was conflicting with `select_related`

kdmccormick committed:
fix: final review comments

kdmccormick committed:
temp: noop

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-33936-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6228458615-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6228458615-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6228458615-tutor-requirements.txt)

kdmccormick committed:
Merge branch 'learning-core-libs-readwrite-runtime' into kdmccormick/noop

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-33936-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6228658180-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6228658180-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6228658180-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-33936-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-33936-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6228808792-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6228808792-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-33936-7daceb-51564656-6228808792-tutor-requirements.txt)
